### PR TITLE
:white_check_mark: (adding jest for unit/integration tests)

### PR DESCRIPTION
**Pull request by MatissJanis**
Adding jest for unit/integration test support.

This is the first step in order to make the API more stable and reliable before starting to add new functionality (https://github.com/actualbudget/actual/discussions/388).

LMK if you want me to change some things around.

### CODE REVIEW

**Comment by MatissJanis** [comment]
Example PR with the new `test` CI task passing:
https://github.com/MatissJanis/actual-server/pull/1

**Comment by MatissJanis** [comment]
Rebased against master branch.

File: app-sync.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          'content-disposition': 'attachment;filename=file-id',
+          'content-type': 'application/octet-stream'
+        })
+      );
+    });
+  });
+
+  // TODO: build the synced-up database support and enable the test cases
+  // Versioning example: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/locking-api-versions.html
+  describe('TBD-date (2022-11-25) version', () => {
```
Comment (j-f1) [inline app-sync.test.js:80]: What is this for?

Comment (MatissJanis) [reply app-sync.test.js]: Originally the idea was to allow customers to get the full synced up database (currently we have the initial database + a lot of CRDT actions, but no way to get everything in a single file from the server).

ButI don't think I'll actually do this anymore, so I removed the todo tests: https://github.com/actualbudget/actual-server/pull/91/commits/6b8b6584f1f458a768851520157ae90ac3641a14

Good catch!

**Review by j-f1 — APPROVED** [review]

**Pull request merged**